src/dash-table/components/Button/ExportButton/index.tsx

```diff
@@ -0,0 +1,46 @@
+import XLSX from 'xlsx';
+import React, {
+    Component
+} from 'react';
+import { IDerivedData, IColumn, IVisibleColumn } from 'dash-table/components/Table/props';
+import { createWorkbook, createHeadings, createWorksheet } from './utils';
+import getHeaderRows from 'dash-table/derived/header/headerRows';
+
+interface IExportButtonProps {
+    export_format: string;
+    virtual_data: IDerivedData;
+    columns: IColumn[] & IVisibleColumn[];
+    export_header: string;
+}
+
+export default class ExportButton extends Component<IExportButtonProps, any> {
```

Comment:
This component looks like a good use-case for a pure functional implementation + React.memo